test/QuestionAPI-test.js

```diff
@@ -294,13 +290,13 @@ describe("Question API Contract", function () {
       const questionIDtx7 = await questionAPI.connect(xmetricaddr1).createQuestion("ipfs://last", 1);
       await questionIDtx7.wait();
 
-      const latestQuestion = await questionAPI.currentQuestionId();
+      const latestQuestion = await bountyQuestion.getMostRecentQuestion();
 
       const allquestionsByState = await questionStateController.getQuestionsByState(new BN(questionState.VOTING), latestQuestion, new BN(1000));
-      expect(allquestionsByState.length).to.equal(8);
+      expect(allquestionsByState.length - 1).to.equal(8);
 
       const allquestionsByStateAgain = await questionStateController.getQuestionsByState(new BN(questionState.VOTING), latestQuestion, new BN(4));
-      expect(allquestionsByStateAgain.length).to.equal(5);
+      expect(allquestionsByStateAgain.length - 1).to.equal(5);
```

Comment:
same as above - if you remove question just change equal number - Better yet we should refactor to have that number be a variable like const allquestionsbyLength = 5; and then 
`expect(allquestionsByStateAgain.length).to.equal(5);`
Does that make sense?

Reply:
Yeah you're right on both, that makes a lot more sense. My brain is tiny